web-client/src/presenter/actions/Reports/BlockedCaseReport/exportCsvBlockedCaseReportAction.ts

```diff
@@ -0,0 +1,62 @@
+import { BlockedFormattedCase } from '@web-client/presenter/computeds/blockedCasesReportHelper';
+import { FORMATS, formatNow } from '@shared/business/utilities/DateHandler';
+
+export type BlockedCsvCase = Pick<
+  BlockedFormattedCase,
+  | 'docketNumber'
+  | 'blockedDateEarliest'
+  | 'caseTitle'
+  | 'status'
+  | 'blockedReason'
+  | 'automaticBlockedReason'
+>;
+
+export const exportCsvBlockedCaseReportAction = ({
+  applicationContext,
+  props,
+}: ActionProps<{
+  blockedCases: BlockedCsvCase[];
+  trialLocation: string;
+}>) => {
+  const { blockedCases, trialLocation } = props;
+  const [city, state] = trialLocation.split(', ');
+  const date = formatNow(FORMATS.MMDDYYYY_UNDERSCORED);
+  const fileName = `Blocked Cases Report - ${city}_${state} ${date}`;
+  const blockedCasesCsvConfiguration = [
+    {
+      displayLabel: 'Docket No.',
+      key: 'docketNumber',
+    },
+    {
+      displayLabel: 'Date Blocked',
+      key: 'blockedDateEarliest',
+    },
+    {
+      displayLabel: 'Case Title',
+      key: 'caseTitle',
+    },
+    {
+      displayLabel: 'Case Status',
+      key: 'status',
+    },
+    {
+      displayLabel: 'Reason',
+      key: 'allReasons',
+    },
+  ];
+
+  const formatString = (s: string | undefined) =>
+    (s || '').split('\n').join(' ').trim();
+
+  const formattedBlockedCases = blockedCases.map(c => ({
+    ...c,
+    allReasons: `${formatString(c.blockedReason)} ${formatString(c.automaticBlockedReason)}`,
+  }));
+
+  const csvString = applicationContext.createCsvString(
```

Comment:
It may be worth considering extracting the formatting logic that prepares the formattedBlockedCases constant into its own function on applicationContext as well so that it can be tested in isolation apart from Cerebral.

Reply:
Because this is such a specific use case of how two strings should be combined(for display purposes) I do not see much utility in extracting it out into a function as I don't know who else would want to use this? But if you have an idea of where else this might be useful I would be down to extract!